File: framework/starter-logging/src/main/java/cn/bestwu/logging/ManagementAuthProperties.java
package cn.bestwu.logging;

import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * 日志访问权限 配置
 *
 * @author Peter Wu
 */
@ConfigurationProperties("summer.management.auth")
public class ManagementAuthProperties {

  private String[] pattern = {};
  /**
   * 访问授权有效时间，单位：秒
   */
  private int maxAge = -1;
  private String username = "madmin";
  private String password;

  public String[] getPattern() {
    return pattern;
  }

  public void setPattern(String[] pattern) {
    this.pattern = pattern;
  }

  public int getMaxAge() {
    return maxAge;
  }

  public void setMaxAge(int maxAge) {
    this.maxAge = maxAge;
  }

  public String getUsername() {
    return username;
  }

  public void setUsername(String username) {
    this.username = username;
  }

  public String getPassword() {
    return password;
  }

  public void setPassword(String password) {
    this.password = password;
  }
}

File: web/src/main/kotlin/top/bettercode/summer/config/ManagementAuthProperties.kt
package top.bettercode.summer.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.util.DigestUtils
import java.nio.charset.StandardCharsets

/**
 * 日志访问权限 配置
 *
 * @author Peter Wu
 */
@ConfigurationProperties("summer.management.auth")
open class ManagementAuthProperties {

    var enabled = true

    //--------------------------------------------
    var pattern = arrayOf<String>()
    var ignorePattern = arrayOf("/health", "/logs/**", DEFAULT_LOGIN_PAGE)

    /**
     * 访问授权有效时间，单位：秒
     */
    var maxAge = 12 * 60 * 60
    var username = "madmin"
    var password: String? = null

    val authKey: String
        //--------------------------------------------
        get() = DigestUtils.md5DigestAsHex(
            "$username:$password".toByteArray(
                StandardCharsets.UTF_8
            )
        )

    companion object {
        const val DEFAULT_LOGIN_PAGE = "/login"

    }
}